ui/admin/app/controllers/scopes/scope/authenticate/method/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { getOwner } from '@ember/application';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+import { notifyError } from 'core/decorators/notify';
+import {
+  TYPE_AUTH_METHOD_PASSWORD,
+  TYPE_AUTH_METHOD_OIDC,
+  TYPE_AUTH_METHOD_LDAP,
+} from 'api/models/auth-method';
+
+export default class ScopesScopeAuthenticateMethodIndexController extends Controller {
+  // =services
+
+  @service session;
+  @service windowManager;
+  @service router;
+
+  // =actions
+
+  /**
+   * Delegates authentication to an authenticator of type matching the current
+   * auth method.  If authentication fails, notifies with an alert.
+   */
+  @action
+  @loading
+  @notifyError(() => 'errors.authentication-failed.title', {
+    sticky: false,
+    catch: true,
+  })
+  async authenticate(authMethod, creds) {
+    const scope = authMethod.scope;
+    const authenticatorName = `authenticator:${authMethod.type}`;
+    const requestCookies = false;
+    switch (authMethod.type) {
+      case TYPE_AUTH_METHOD_PASSWORD:
+      case TYPE_AUTH_METHOD_LDAP:
+        await this.session.authenticate(
+          authenticatorName,
+          creds,
+          requestCookies,
+          { scope, authMethod },
+        );
+        this.router.transitionTo('index');
+        break;
+      case TYPE_AUTH_METHOD_OIDC:
+        await this.startOIDCAuthentication(authenticatorName, {
+          scope,
+          authMethod,
+        });
+        this.router.transitionTo('scopes.scope.authenticate.method.oidc');
+        break;
+    }
+  }
+
+  /**
+   * Kicks off the OIDC auth flow by opening a new window to the provider.
+   * @param {string} authenticatorName
+   * @param {object} options
+   */
+  async startOIDCAuthentication(authenticatorName, options) {
+    const oidc = getOwner(this).lookup(authenticatorName);
+    // TODO: delegate this call from the session service so that we don't have
+    // to look up the authenticator directly
```

Comment:
This is an existing TODO and I'm not quite sure what it's referring to. Anyone have any thoughts in regards to this?

Reply:
I'm not exactly sure but maybe because OIDC is slightly different and doesn't call the session service directly?

Reply:
ill create a ticket to spend some more time on this particular todo